plugins/modem-manager/fu-firehose-updater.c

```diff
@@ -540,44 +540,39 @@ fu_firehose_updater_send_program_file(FuFirehoseUpdater *self,
 				      gsize sector_size,
 				      GError **error)
 {
-	g_autoptr(FuChunk) chk_last = NULL;
 	g_autoptr(FuChunkArray) chunks = fu_chunk_array_new_from_bytes(program_file,
 								       FU_CHUNK_ADDR_OFFSET_NONE,
 								       FU_CHUNK_PAGESZ_NONE,
 								       payload_size);
 
-	/* last block needs to be padded to the next sector_size,
-	 * so that we always send full sectors */
-	chk_last = fu_chunk_array_index(chunks, fu_chunk_array_length(chunks) - 1, error);
-	if (chk_last == NULL)
-		return FALSE;
-	if ((fu_chunk_get_data_sz(chk_last) % sector_size) != 0) {
-		g_autoptr(GBytes) padded_bytes = NULL;
-		gsize padded_sz = sector_size * (fu_chunk_get_data_sz(chk_last) / sector_size + 1);
-
-		padded_bytes = fu_bytes_pad(fu_chunk_get_bytes(chk_last), padded_sz);
-		fu_chunk_set_bytes(chk_last, padded_bytes);
-
-		g_return_val_if_fail(fu_chunk_get_data_sz(chk_last) == padded_sz, FALSE);
-	}
 	for (guint i = 0; i < fu_chunk_array_length(chunks); i++) {
 		g_autoptr(FuChunk) chk = NULL;
+		g_autoptr(GBytes) block = NULL;
+		g_autoptr(GBytes) block_padded = NULL;
+		gsize block_padded_sz;
 
 		/* prepare chunk */
 		chk = fu_chunk_array_index(chunks, i, error);
 		if (chk == NULL)
 			return FALSE;
 
+		block = fu_chunk_get_bytes(chk);
+		/* block needs to be padded to the next sector_size,
+		 * so that we always send full sectors */
+		block_padded_sz =
+		    ((g_bytes_get_size(block) + sector_size - 1) / sector_size) * sector_size;
+		block_padded = fu_bytes_pad(block, block_padded_sz);
```

Comment:
Is this different from calling `block_padded = fu_bytes_pad(block, sector_size)`?

Reply:
There is a difference between block size ('payload size') and sector size for the Sahara protocol. Block size is usually 8192 bytes, while sector size is mostly 2048. If we have 1000 bytes remaining, we only have to pad up to 2048, not 8192.

Reply:
Okay, got it, thanks.